src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Address.xml

```diff
@@ -30,6 +30,10 @@
                 <attribute name="method">GET</attribute>
                 <attribute name="path">/shop/addresses/{id}</attribute>
             </itemOperation>
+            <itemOperation name="shop_delete">
+                <attribute name="method">DELETE</attribute>
+                <attribute name="path">/shop/addresses/{id}</attribute>
```

Comment:
we have address book only on shop context są instead custom `path` you can add parameter `route_prefix`